v2/mfa/backend/second-factor.mdx

```diff
@@ -657,7 +793,86 @@ Passwordless.init({
 TODO
 </TabItem>
 <TabItem value="python">
-TODO
+
+```python
+from supertokens_python.recipe.passwordless.interfaces import APIInterface, APIOptions
+from typing import Union, Dict, Any, Optional
+from supertokens_python.recipe.session.asyncio import get_session
+from supertokens_python.recipe.usermetadata.asyncio import get_user_metadata
+from supertokens_python.recipe.passwordless.asyncio import get_user_by_id
+
+
+def override_passwordless_apis(original_implementation: APIInterface):
+    original_consume_code_post = original_implementation.consume_code_post
+    original_create_code_post = original_implementation.create_code_post
+
+    async def create_code_post(
+        email: Union[str, None],
+        phone_number: Union[str, None],
+        api_options: APIOptions,
+        user_context: Dict[str, Any],
+    ):
+        # This API is called to send an OTP
+        # We want to make sure that the OTP being generated is for the
+        # same number that belongs to this user.
+
+        # A session should already exist since this should be called after the first
+        # factor is completed
+        _session = await get_session(api_options.request)
+        assert _session is not None
+
+        # We try to get the phone number associated with this user. It will be
+        # defined if this is a sign in attempt, in which case, we will check that
+        # it is equal to the input phone number
+        user_metadata = await get_user_metadata(_session.get_user_id())
+        user_metadata_phone_number: Optional[str] = None
+        if user_metadata.metadata.get("passwordlessUserId"):
+            # the flow will come here during a login attempt, since we
+            # associate the passwordless userId to the user on sign up
+            passwordless_user_info = await get_user_by_id(
+                user_metadata.metadata["passwordlessUserId"], user_context
+            )
+            if passwordless_user_info is not None:
+                user_metadata_phone_number = passwordless_user_info.phone_number
+
+        if user_metadata_phone_number is not None:
+            # this means we found a phone number associated to this user
+            # we will check if the input phone number is the same as this one.
+            if phone_number is not None and phone_number != user_metadata_phone_number:
```

Comment:
This condition is not the same as `if (!("phoneNumber" in input) || input.phoneNumber !== phoneNumber) {`

What you want to do is `if phone_number is None or phone_number != user_metadata_phone_number:`